dashboard/app/assets/javascripts/testExecutor.service.js

```diff
@@ -0,0 +1,73 @@
+(function() {
+  'use strict';
+
+  angular.module('qat-dashboard').service('testExecutor', [
+    'testExecsFactory',
+    function(testExecsFactory) {
+      var service = this;
+
+      service.all = {};
+      service.new = newTestExec;
+      service.delete = deleteTestExec;
+
+      var testExecUpdateIntervalPeriod = 5; // seconds
+
+      // Populate all
+      testExecsFactory.index(null, function(testExecsList) {
+        testExecsList.forEach(function(testExec) {
+          service.all[testExec.uuid] = testExec;
+          if (testExec.status !== 'DONE') {
+            updateTestExec(testExec.uuid);
+          }
+        });
+      });
+
+      function newTestExec() {
+        testExecsFactory.create(null, function(testExec) {
+          service.all[testExec.uuid] = testExec;
+          updateTestExec(testExec.uuid);
+        }, function(httpResponse) {
+          console.error('error while requesting new testExec: ' + httpResponse.code);
+        });
+      }
+
+      function updateTestExec(uuid) {
+        var updateInterval = setInterval(function() {
+          testExecsFactory.show({id: uuid}, function(updatedTestExec) {
+            if (updatedTestExec.status === 'DONE' || updatedTestExec.status === 'ERRORED') {
+              clearInterval(updateInterval);
+            }
+            service.all[uuid] = updatedTestExec;
+          });
+        }, testExecUpdateIntervalPeriod * 1000);
+      }
+
+      function deleteTestExec(uuid) {
+        testExecsFactory.delete({id: uuid}, function(testExec) {
+          delete service.all[testExec];
+        });
+      }
+    }
+  ]);
+
+  angular.module('qat-dashboard').factory('testExecsFactory', [
```

Comment:
@christianvuerings I placed testExecutorFactory in this file because it was supposed to be used only in the testExecutor service. I have deleted the factory and moved its functionality into the testExecutor service.